test/common/run-tests

```diff
@@ -67,83 +66,84 @@ def print_test(test, print_tap=True, retry_reason=""):
                 line = line[e.characters_written:]
                 time.sleep(0.1)
 
+    if retry_reason:
+        retry_reason = " " + retry_reason
+    if skip_reason:
+        skip_reason = " " + skip_reason
+
     if not print_tap:
-        print(retry_reason)
+        print(retry_reason + skip_reason)
         flush_stdout()
         return
 
     print()  # Tap needs to start on a separate line
     if test.process.returncode == 0:
-        print("ok " + test_name(test) + retry_reason)
-    elif test.process.returncode == 77 or b"# SKIP " in test.output:
-        # If the test was skipped, add the last line (which contains the reason
-        # for the skip) to the result
-        print("ok {0} {1}{2}".format(test_name(test),
-                                     test.output.splitlines()[-1].strip().decode() if test.process.returncode == 77 else "",
-                                     retry_reason))
+        print("ok {0}{1}".format(test_name(test), retry_reason))
+    elif skip_reason:
+        print("ok {0}{1}".format(test_name(test), skip_reason))
     else:
-        print("not ok " + test_name(test) + retry_reason)
+        print("not ok {0}{1}".format(test_name(test), retry_reason))
     flush_stdout()
 
 
 def finish_test(opts, test, affected_tests):
     """Returns if a test should retry or not
 
-    Call test-policy on the test's output, print if needed.
+    Call test-failure-policy on the test's output, print if needed.
 
     Return (retry_reason, exit_code). retry_reason can be None or a string.
     """
 
     affected = any([test.command[0].endswith(t) for t in affected_tests])
+    retry_reason = ""
+    skip_reason = ""
 
     # Try affected tests 3 times
     if test.process.returncode == 0 and affected and test.retry_when_affected and test.retries < 2:
-        retry_reason = b"test affected tests 3 times"
+        retry_reason = "test affected tests 3 times"
         test.retries += 1
-        print_test(test, not opts.list, " # RETRY {0} ({1})".format(test.retries, retry_reason.decode("utf-8")))
+        print_test(test, not opts.list, "# RETRY {0} ({1})".format(test.retries, retry_reason))
         return retry_reason, 0
-    elif test.process.returncode in [0, 77]:
-        print_test(test, not opts.list)
+
+    # If test is being skipped pick up the reason
+    if test.process.returncode == 77:
+        lines = test.output.splitlines()
+        skip_reason = lines[-1].strip().decode("utf-8")
+        test.output = b"\n".join(lines[:-1])
```

Comment:
I thought we now wouldn't have to do this split/parse/join dance any more -- the entire output should be the reason, no?

Reply:
Well yeah, this one is special case: https://github.com/cockpit-project/cockpit/blob/master/test/common/testlib.py#L1579

But it is just "last line when returned 77" which is rather safe compered to previous regexes.

Reply:
I see, thanks!